1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [1]:
import numpy as np
from scipy import stats

In [2]:
stats.norm.ppf(0.025)

-1.9599639845400545

In [3]:
80 + stats.norm.ppf(0.025) * (16 / np.sqrt(256))

78.04003601545995

In [4]:
80 - stats.norm.ppf(0.025) * (16 / np.sqrt(256))

81.95996398454005

Ответ: (78.04003601545995,  81.95996398454005)

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95

In [5]:
t = stats.t.ppf(0.025, 9)
t

-2.262157162740992

In [6]:
data = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
mean_data =  np.mean(data)
mean_data

6.590000000000001

In [7]:
st = np.std(data, ddof=1)
st

0.4508017549014448

In [8]:
mean_data + t * (st / np.sqrt(len(data)))

6.267515851415713

In [9]:
mean_data - t * (st / np.sqrt(len(data)))

6.912484148584288

Ответ: (6.267515851415713, 6.912484148584288)

3. Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170 Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175 Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [10]:
growth_daughters = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
growth_mothers = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
delta = np.mean(growth_daughters) - np.mean(growth_mothers)
delta

-1.9000000000000057

In [11]:
d_both_groups = (np.var(growth_daughters, ddof=1) + np.var(growth_mothers, ddof=1)) / 2
d_both_groups

75.58333333333334

In [12]:
s_delta = np.sqrt(d_both_groups/len(growth_daughters) + d_both_groups/len(growth_mothers))
s_delta

3.8880157750022915

In [13]:
t = stats.t.ppf(0.025, 2*(len(growth_daughters) -1))
t

-2.10092204024096

In [14]:
delta + t * s_delta

-10.068418034506857

In [15]:
delta - t * s_delta

6.268418034506846

Ответ: (-10.068418034506857, 6.268418034506846)